src/mym.cpp

```diff
@@ -888,6 +894,120 @@ void mexFunction(int nlhs, mxArray*plhs[], int nrhs, const mxArray*prhs[]) {
+            
+            char** pd = (char**)mxCalloc(nac, sizeof(char*)); // output of serialized variable, for each placeholder
+            size_t* plen = (size_t*)mxCalloc(nac, sizeof(size_t)); // length of bytes in the serialized variable, for each place holder
+            
+            // now we have the correct number of placeholders
+            // read the types and in-string arguments
+            ps = (unsigned*)mxCalloc(nac, sizeof(unsigned));
+            pa = (char**)mxCalloc(nac, sizeof(char*));
+            pec = (bool*)mxCalloc(nac, sizeof(bool));
+            pf = (pfserial*)mxCalloc(nac, sizeof(pfserial));
+            for (unsigned i = 0; i<nac; i++) {
+                // placeholder function+control ph type+control arg type
+                getSerialFct(po[i]+strlen(PH_OPEN), prhs[jarg+i+1], pf[i], pec[i]); // first parameter points to the first character of placeholder
+                // placeholder size in bytes
+                ps[i] = (unsigned)(pc[i]-po[i]+strlen(PH_OPEN));
+                // placeholder in-string argument
+                pa[i] = po[i]+strlen(PH_OPEN)+1; // e.g. i in {Si}
+                *(pc[i]) = 0;
+                // we can add a termination character in the query,
+                // it will be processed anyway and we already have the original query length
+            }
+            // serialize
+            uLongf cmp_buf_len = 10000;
+            Bytef* pcmp = (Bytef*)mxCalloc(cmp_buf_len, sizeof(Bytef));
+            size_t nb = 0;
+            for (unsigned i = 0; i<nac; i++) {
+                // serialize individual field
+                pd[i] = pf[i](plen[i], prhs[jarg+i+1], pa[i], true); // serialize the variable field
+                if (pec[i] && (plen[i]>MIN_LEN_ZLIB)) { // compress if needed
+                    // compress field
+                    const uLongf max_len = compressBound(plen[i]);
+                    if (max_len>cmp_buf_len){
+                        pcmp = (Bytef*)mxRealloc(pcmp, max_len);
+                        cmp_buf_len = max_len;
+                    }
+                    uLongf len = cmp_buf_len;
+                    if (compress(pcmp, &len, (Bytef*)pd[i], plen[i])!=Z_OK)
+                        mexErrMsgTxt("Compression failed");
+                    const float cmp_rate = plen[i]/(LEN_ZLIB_ID+1.f+sizeof(_uint64)+len);
+                    if (cmp_rate>MIN_CMP_RATE_ZLIB) {
+                        const size_t len_old = plen[i];
+                        plen[i] = LEN_ZLIB_ID+1+sizeof(_uint64)+len;
+                        pd[i] = (char*)mxRealloc(pd[i], plen[i]);
+                        memcpy(pd[i], (const char*)ZLIB_ID, LEN_ZLIB_ID);
+                        *(pd[i]+LEN_ZLIB_ID) = 0;
+
+                        *((_uint64*)(pd[i]+LEN_ZLIB_ID+1)) = (_uint64) len_old;
+                        memcpy(pd[i]+LEN_ZLIB_ID+1+sizeof(_uint64), pcmp, len);
+                        //BUG: *((_uint64*)(pd[i]+LEN_ZLIB_ID+1+sizeof(_uint64))) = (_uint64) len;

```

Comment:
what is this for?

Reply:
Again, it is copied from original code, I have now forgotten what this was for, you can see that the code is extremely difficult to follow